src/main/java/Duke.java

```diff
@@ -1,10 +1,132 @@
+import java.io.IOException;
+import java.util.Scanner;
+
+/**
+ * Duke is a chatbot that allows users to send input to perform tasks
+ */
 public class Duke {
-    public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+    private Storage storage;
+    private TaskList taskList;
+    private Ui ui;
+
+    /**
+     * Creates a new Duke chatbot and load tasks from storage.
+     *
+     * @param filePath path to the storage file to load tasks from
+     */
+    public Duke(String filePath) throws IOException {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            // Load task list from saved file
+            taskList = new TaskList(storage.load());
+        } catch (IOException e) {
+            // Create task list
+            ui.showLoadingError();
+            taskList = new TaskList();
+        }
+    }
+
+    /**
+     * Runs main conversation loop with Duke chatbot.
+     */
+    public void run() {
+        // Introduction messages
+        System.out.println("Hello! I'm Duke! I'm a chatbot-based To-Do list manager.");
+        System.out.println("My available commands are: todo, deadline, event, done, list, delete, bye");
+        System.out.println("What can I do for you today? :)");
+
+        // Main conversation loop
+        Scanner sc = new Scanner(System.in);
+        boolean isSpeaking = true;
+        while (isSpeaking) {
+
+            // Process user input
+            String userInput = sc.nextLine();
+            String[] userTokens = userInput.split(" ");
+            String userCommand = userTokens[0];
+
+            // Validate command
+            try {
+                Parser.validateCommand(userCommand);
+            } catch (DukeException e) {
+                System.out.println("Sorry, that looks like an invalid command! " + e.getMessage());
+            }
+
+            switch (userCommand) {
+                // Exit the program
+                case "bye":
```

Comment:
Same as hanweic53 has pointed out. It's a minor issue, but it might be good to take note!